packages/test/test-end-to-end-tests/src/test/handleValidation.spec.ts

```diff
@@ -0,0 +1,348 @@
+	// 		const change: Change = Change.setPayload(nodeId, handle);
+	// 		treeRoot.applyEdit(change);
+	// 	},
+	// },
+	{
+		type: registerId,
+		storeHandle: async (defaultDataStore, handle) => {
+			const registerRoot =
+				await defaultDataStore.getSharedObject<IConsensusRegisterCollection<FluidObject>>(
+					registerId,
+				);
+			await registerRoot.write("B", handle);
+		},
+		readHandle: async (defaultDataStore) => {
+			const registerRoot =
+				await defaultDataStore.getSharedObject<IConsensusRegisterCollection<FluidObject>>(
+					registerId,
+				);
+			return registerRoot.read("B");
+		},
+	},
+	{
+		type: queueId,
+		storeHandle: async (defaultDataStore, handle) => {
+			const queueRoot =
+				await defaultDataStore.getSharedObject<IConsensusOrderedCollection>(queueId);
+			await queueRoot.add(handle);
+		},
+		readHandle: async (defaultDataStore) => {
+			let handle2: unknown;
+			const queueRoot =
+				await defaultDataStore.getSharedObject<IConsensusOrderedCollection>(queueId);
+			const callback: ConsensusCallback<IFluidHandle> = async (value) => {
+				handle2 = value;
+				return ConsensusResult.Complete;
+			};
+			await queueRoot.waitAndAcquire(callback);
+			return handle2;
+		},
+	},
+	// {
+	// 	type: migrationShimId,
+	// 	storeHandle: async (defaultDataStore, handle) => {
+	// 		const migrationShimRoot = await defaultDataStore.getSharedObject<MigrationShim>(migrationShimId);
+	// 		const tree = migrationShimRoot.currentTree as LegacySharedTree;
+	// 		const legacyNodeId: TraitLabel = "inventory" as TraitLabel;
+
+	// 		const handleNode: BuildNode = {
+	// 			definition: legacyNodeId,
+	// 			traits: {
+	// 				handle,
+	// 			},
+	// 		};
+	// 		tree.applyEdit(
+	// 			Change.insertTree(
+	// 				handleNode,
+	// 				StablePlace.atStartOf({
+	// 					parent: tree.currentView.root,
+	// 					label: legacyNodeId,
+	// 				}),
+	// 			),
+	// 		);
+
+	// 		const rootNode = tree.currentView.getViewNode(tree.currentView.root);
+	// 		// eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+	// 		const nodeId = rootNode.traits.get(legacyNodeId)![0];
+	// 		const change = Change.setPayload(nodeId, handle);
+	// 		tree.applyEdit(change);
+	// 	},
+	// },
+];
+
+describeCompat("handle validation", "NoCompat", (getTestObjectProvider, apis) => {
+	for (const handle of handleFns) {
+		it(`store handle in dds: ${handle.type}`, async () => {
+			const { container1, provider, testContainerConfig } = await setup(
+				getTestObjectProvider,
+				apis,
+			);
+			let idA: string;
+			let seq: number;
+			{
+				const defaultDataStore = (await container1.getEntryPoint()) as ITestFluidObject;
+				const runtime = defaultDataStore.context.containerRuntime;
+				idA = defaultDataStore.context.id;
+
+				const dataStoreB = await runtime.createDataStore(["default"]);
+				const dataObjectB = (await dataStoreB.entryPoint.get()) as ITestFluidObject;
+
+				await handle.storeHandle(defaultDataStore, dataObjectB.handle);
```

Comment:
These tests do fail without calling bindHandles and without Tony's fix. I also might have gotten confused on the test scenario, but looking at [the commit we protoyped](https://github.com/microsoft/FluidFramework/compare/main...markfields:FluidFramework:dds/detached-bind?expand=1#diff-d86be5ae38405e0e981d4a47b66969c1f243f9e14a6d001eab5af75cce4bc630R902) it seems like the test case was:
1. DefaultDataStore is attached (of course). Create DataStoreB (it's detached).
2. Store a handle to DataObjectB in the root of DefaultDataStore (effectively storing it in a map)
I planned on doing a follow up anyway, so I can add the second test case you point out in that PR.

Reply:
That's great to know you verified the test coverage.

As for the other test case - you're right that this is what the test was doing.  But when we were debugging, it was the test fluid object initialization for DataStoreB that was storing handles between DDSes, and _that_ was the part that was causing multiple attach ops.  Super subtle.

My concern is that all these test cases are actually just testing Map under the hood, because it's in DataStoreB's root Map where the handles are stored that actually matter.

cc @anthony-murphy as FYI